test/com/amazon/ion/streaming/ReaderTest.java

```diff
@@ -178,159 +179,121 @@ public void testSymbolValueOnNonSymbol()
         }
     }
 
-    @Test
-    public void testReadingDecimalAsInteger()
-    {
-        // decimal value <space> int conversion
-        read(
-            "0. 0 " +
-            "-2147483648. -2147483648 " + // Min int
-            "2147483647. 2147483647 " // Max int
-        );
-        while (in.next() != null)
-        {
-            int actual = in.intValue();
+    int readInt(String text) {
+        IonReader r = myReaderMaker.newReader(system(), text);
```

Comment:
I recommend just using `read(text)` and using `in` as the reader, here and in the other new `read*()` methods.